### ISSUE TITLE
Slicing an Orbit with multiple orbits?

### ISSUE THREAD

**Issue by adrn** [GalacticDynamics/galax#742]
I used `evaluate_orbit` with a `PhaseSpacePosition` with shape `(2,)`, then sliced the output orbit:
```python
pot = gp.NFWPotential.from_M200_c(Quantity(1e12, "Msun"), 10.0, units="galactic")
w0 = gc.PhaseSpacePosition(
    q=Quantity([[5.0, 0.0, 0.0], [5.05, 0.0, 0.0]], "kpc"),
    p=Quantity([[0.0, 200.0, 0.0], [0, 200, 0]], "km/s"),
)
ts = Quantity(jnp.linspace(0.0, 2.0, 128), "Gyr")
orbit = gd.evaluate_orbit(pot, w0, ts)
orbit.shape, orbit[0].shape, type(orbit[0])
```
Output:
```
(2, 128), (128,), galax.coordinates._src.pscs.single.PhaseSpaceCoordinate
```
Shouldn't this slice return another `Orbit` object, since I'm not slicing on the time axis?

**Comment by adrn** [comment]
Also: if I do `orbit.plot(x="x", y="y")`, I get this:
![Image](https://github.com/user-attachments/assets/aec24e51-7ad0-45b2-b094-7cb5a349e6f9)

But I think we should iterate over the other axis so all points from the same orbit are plotted as a continuous curve.

**Comment by nstarman** [comment]
> Shouldn't this slice return another Orbit object, since I'm not slicing on the time axis?

Yes.

**Comment by adrn** [comment]
Cool -- I'll work on this, I just wanted to open the issue to remember!

**Comment by nstarman** [comment]
There's also a bug with ND slicing of orbits

```python
w = gc.PhaseSpaceCoordinate(
    q=u.Quantity([[[8, 0, 0], [9, 0, 0]]], "kpc"),
    p=u.Quantity([0, 220, 0], "km/s"),
    t=u.Quantity(0, "Myr"),
)
usys = u.unitsystem("kpc", "Msun", "Myr", "rad")
pot = gp.NFWPotential(
    m=u.Quantity(1.77e12, "Msun"),
    r_s=u.Quantity(20, "kpc"),
    units=usys,
)
orbit = gd.evaluate_orbit(pot, w, t=u.Quantity(jnp.linspace(0, 20, 3), "Myr"))

orbit = jax.block_until_ready(orbit)
print(orbit.shape)
# (1, 2, 3)

orbit[0, 0]
# IndexError
```
